crabs/detection_tracking/detection_utils.py

```diff
@@ -226,65 +229,243 @@ def calculate_iou(box1, box2) -> float:
+            if iou > iou_threshold and iou > best_iou:
+                best_iou = iou
+                best_match = j
+        if best_match is not None:
+            gt_boxes[best_match] = None
+        else:
+            false_alarms += 1
+
+    missed_detections = 0
+    for box in gt_boxes:
+        if box is not None and not np.all(np.isnan(box)):
+            missed_detections += 1
+
+    num_switches = count_identity_switches(prev_frame, tracked_boxes)
+    mota = 1 - (missed_detections + false_alarms + num_switches) / total_gt
+    return mota
+
+
+def get_ground_truth_data(gt_dir: str) -> list:
+    """
+    Extract ground truth bounding box data from a CSV file.
+
+    Parameters:
+    -----------
+    gt_dir : str
+        The path to the CSV file containing ground truth data.
+
+    Returns:
+    --------
+    list:
+        A list containing ground truth bounding box data organized by frame number.
+    """
+    ground_truth_data = []
+    max_frame_number = 0
+
+    # Open the CSV file and read its contents line by line
+    with open(gt_dir, "r") as csvfile:
+        csvreader = csv.reader(csvfile)
+        next(csvreader)  # Skip the header row
+        for row in csvreader:
+            # Extract relevant information from each row
+            filename = row[0]
+            region_shape_attributes = json.loads(row[5])
+            region_attributes = json.loads(row[6])
+
+            # Extract bounding box coordinates and object ID
+            x = region_shape_attributes["x"]
+            y = region_shape_attributes["y"]
+            width = region_shape_attributes["width"]
+            height = region_shape_attributes["height"]
+            track_id = region_attributes["track"]
+
+            # Compute the frame number from the filename
+            frame_number = int(filename.split("_")[-1].split(".")[0])
+            frame_number = frame_number - 1
+
+            # Update max_frame_number
+            max_frame_number = max(max_frame_number, frame_number)
+
+            # Append the extracted data to the list
+            ground_truth_data.append(
+                {
+                    "frame_number": frame_number,
+                    "x": x,
+                    "y": y,
+                    "width": width,
+                    "height": height,
+                    "id": track_id,
+                }
+            )
+
+    # Initialize a list to store the ground truth bounding boxes for each frame
+    gt_boxes_list = [np.array([]) for _ in range(max_frame_number + 1)]
+
+    # Organize ground truth data into gt_boxes_list
+    for data in ground_truth_data:
+        frame_number = data["frame_number"]
+        bbox = np.array(
+            [
+                data["x"],
+                data["y"],
+                data["x"] + data["width"],
+                data["y"] + data["height"],
+                data["id"],
+            ],
+            dtype=np.float32,
+        )
+        gt_boxes_list[frame_number] = (
```

Comment:
This bit took me a while, but I think it can be more clear if the if-case is spelled out. 

Is the goal to return a list of arrays, each array collecting the tracked bounding boxes for one frame? Maybe this is good to add to the docstrings. I would also spell out the if-case:

```
 # if there are bounding boxes defined for the array at this frame number:
 # vertically stack the new bounding box
if gt_boxes_list[frame_number].size: 
	gt_boxes_list[frame_number] = np.vstack([gt_boxes_list[frame_number], bbox])
 # otherwise define the first row of the array for this frame number
else:
	gt_boxes_list[frame_number] = bbox
```

Question: If the frame number is 1-based, does that mean that `gt_boxes_list`'s first element is an empty array?

Reply:
I re-index the frame_number variable to 0
`frame_number = int(filename.split("_")[-1].split(".")[0]) - 1`